Makefile

```diff
@@ -35,6 +35,7 @@ GOOS ?= $(shell go env GOOS)
 GOARCH ?= $(shell go env GOARCH)
 GOBUILD=CGO_ENABLED=0 installsuffix=cgo go build -trimpath
 GOTEST=go test -v $(RACE)
+GOTESTREPORT=gotestsum --junitfile test-report.xml -- -v $(RACE)
```

Comment:
perhaps instead of adding new var we can simply modify GOTEST. One nice thing about gotestsum is that it colors the output, which we had to do manually
```
GOTESTREPORT=gotestsum --format testname --junitfile test-report.xml -- -v $(RACE)
```

Reply:
I tried it first but it seems several targets uses GOTEST without running install-ci so almost all the PR checks failed on that attempt.

Reply:
those targets should be easy to fix though. Or we could make GOTEST point to a shell script that will ensure the module is installed before executing